### ISSUE TITLE
Cannot access Internet

### ISSUE THREAD

**Issue by turb0chrg** [MicrosoftDocs/Virtualization-Documentation#971]
Thanks for this, transparent networking between guests and other Azure VMs is a must. Having to create PAT rules using NAT isn't very scalable.

However, following the article I cannot get the nested guests to be able to access the Internet, Azure DNS, etc. The nested guests are able to access Azure VMs and vice-versa without issue. The host can access the Internet without issue. Also, a couple of suggestions for the article:
1. It doesn't mention that IP Forwarding is required to be enabled on the Hyper-V host in Azure.
2. The last step of the "Configuring Remote Access" should say to use the "Private interface connected to private network" for Interface Type.

Maybe that's my issue - I never see any translated packets in the NAT section of RRAS. Thanks!

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 07fc9a55-6666-e505-15f1-015b6752d0c1
* Version Independent ID: cf12c1c7-d8f2-8bb1-b8dd-9cbe81b60d03
* Content: [Configuring Nested VMs to Communicate Directly with Resources in an Azure Virtual Network](https://docs.microsoft.com/en-us/virtualization/hyper-v-on-windows/user-guide/nested-virtualization-azure-virtual-network)
* Content Source: [virtualization/hyper-v-on-windows/user-guide/nested-virtualization-azure-virtual-network.md](https://github.com/Microsoft/Virtualization-Documentation/blob/master/virtualization/hyper-v-on-windows/user-guide/nested-virtualization-azure-virtual-network.md)
* Service: **windows-10-hyperv**
* Product: **windows-10-hyperv**
* GitHub Login: @mrajess

**Comment by mrajess** [comment]
Yea, that's my fault. The original article didn't account for internet connectivity and just enabled nested VMs to communicate with Azure VMs directly. That was due to the initial scenario that I came up with this for. I then tried hamstringing in internet connectivity, didn't test everything and here we are. I've now solved for this. I'm going to redo the write up today, but you'll end up needing two NICs and two Subnets for the Hyper-V server. 

One NIC will sit on a Subnet, let's call it NAT Subnet, and will be configured in RRAS to do NAT. This NIC will also be configured with a default gateway. This will give the VMs internet connectivity.

The second NIC will be on a subnet that just allows for communication between the nested VMs and the servers in Azure. This NIC will not have a default gateway. 

Basically, one NIC handles internet, and the other handles LAN. Only way I could get this to work, but it does work! Apologies for any frustration. I only get to work on this in my free time and I kind of rushed the edit out. :)

**Comment by turb0chrg** [comment]
Thanks for your response!

**Comment by tscottg76** [comment]
Hi. Has this document been updated based on your last comment? Basically I have everything setup where my nested VM in Hyper-v can access other Azure VMs and the Internet. But my Azure VM's cannot access my nested VM's on Hyper-v. 
My Azure VM that is a hyper-v host only has one nic and the one virtual switch.

Also, instead of creating route tables in the Azure portal for the Azure VM to access the nested VMs, can you also manually create the routes on each of the Azure VMs from a command prompt e.g. route add 10.0.1.0 mask 255.255.255.0 10.0.0.4 ?

**Comment by davidhoward74** [comment]
Can you advise when this re-write will be available?  I have been trying to follow previous and updated guidance on how to enable communications between nested VM's and other machines on the vNet in Azure and on-prem, but without success.

**Comment by mrajess** [comment]
https://github.com/MicrosoftDocs/Virtualization-Documentation/blob/master/virtualization/hyper-v-on-windows/user-guide/nested-virtualization-azure-virtual-network.md

It hasn't been merged into Live yet, but is available at the Master linked above.

**Comment by davidhoward74** [comment]
Thanks for the link.. However, I have been unable to get this working successfully. At what point does the fourth Subnet described in the 'End State Configuration' come into play? Previous steps, do not seem to refer to this.
Would you be able to provide a diagram to show how the subnet and routing config should work?

**Comment by Flodu31** [comment]
Hi All,

Everything work for me based on the github article, except Internet. I've no internet access for my nested VMs.

Florent

**Comment by ramraj4k** [comment]
Has Internet connectivity is accessible on Nested VM inside the Hyper-v Console ? ...noticed still there is an issue to access the internet. Any advise ??

**Comment by tkdaymond** [comment]
Does anyone have any general troubleshooting steps for this besides double-checking the configuration (which I've done a couple of times now)? 

My nested VM (in vNet2) has access to the Hyper-V host, but not to anything else. It cannot access the Internet, Azure DNS or other Azure VMs. My other VMs are on a different vNet (vNet1), but I have a working GatewaySubnet transferring traffic between the two vNets.

Also, does anyone have guidance on the two "variable" areas of the procedure? I'm referring to the creation of additional routes on Ethernet 2 in addition to Route 2 (in the Configuring Route Access section) AND the association of additional subnets in the "Configuring the route table" section. For example, I guess I should add routes and associate subnets for subnets in my other vNet (vNet1) since I want them to be able to access the nested VMs in vNet2 as well.

Regarding the troubleshooting, should I expect to see any values for the Ethernet interface in the NAT node of the IPv4 node in RRAS? I don't. They are zero and that makes me think that the NAT in RRAS is not functioning.

**Comment by nadiyazac** [comment]
This fixed my Internet issue.
Please run the following on the host (hyper-V)
New-Netnat -Name "NestedSwitch" -InternalIPInterfaceAddressPrefix 10.0.2.0/24

**Comment by capics** [comment]
I've no issues connecting to the Internet. To ping Internet addresses like 8.8.8.8 you've to add a Public IP address to the Primary NIC of the Hyper-V Host.

I do have a problem though. I'm unable to ping the nested VM (DHCP IP address 10.0.2.2) from anywhere accept from the Hyper-V host?!

Remark: The first two bullet points in the High Level Overview state that the nested VM has two NIC's. This is incorrect I think, because the host has two NIC's, not the nested VM. Is that correct?

**Comment by nanosys-cloud** [comment]
Its works for me, I just add NetNat on host VM and the VM in hyper-v can access internet now.

**Comment by janissimsons** [comment]
@nadiyazac thank you! That was the missing command for me as well! Nested VM can now connect to the internet.
@mrajess why isn't nadiyazac comment part of your guide? It is essential for nested VM network to work properly, otherwise it does not work, I followed your guide to the letter, double checking everything and initially I had the same problem as everyone else above. But once I ran 
 New-Netnat -Name "NestedSwitch" -InternalIPInterfaceAddressPrefix 10.0.2.0/24
then everything started working.

**Comment by robinharwood** [comment]
We hugely appreciate your contribution, feedback, and support to our technical documentation.

We’ve phased out GitHub Issues on this repository as the feedback mechanism for content and have replaced it with a new feedback system. For more information see, [Content User Feedback](https://aka.ms/ContentUserFeedback). As part of this rollout, we are migrating your issue where our team will use your comments to improve the documentation experience. Unfortunately, this does mean you won't be able to access this issue after the migration is complete.

All issues will be migrated by Thursday 29th August 2024.

Once again, thank you for your valuable feedback, we hope you’ll continue to contribute using our new feedback system.
